libs/dyn/dynloc/locations.go

```diff
@@ -0,0 +1,185 @@
+package dynloc
+
+import (
+	"fmt"
+	"path/filepath"
+	"slices"
+	"sort"
+
+	"github.com/databricks/cli/libs/dyn"
+	"golang.org/x/exp/maps"
+)
+
+const (
+	// Version is the version of the location information structure.
+	// Increment if the structure changes.
+	Version = 1
+)
+
+// Locations is a structure that holds location information for (a subset of) a [dyn.Value] value.
+type Locations struct {
+	// Version is the version of the location information.
+	Version int `json:"version"`
+
+	// Files is a list of file paths.
+	Files []string `json:"files"`
+
+	// Locations maps the string representation of a [dyn.Path] to a list of 3-tuples that represent the index
+	// of the file in the [Files] array, followed by the line and column number.
+	// A single [dyn.Path] can have multiple locations (e.g. the effective location and original definition).
+	Locations map[string][][]int `json:"locations"`
+
+	// fileToIndex maps file paths to their index in the [Files] array.
+	// This is used to avoid duplicate entries in the [Files] array and keep the
+	// map with locations as compact as possible.
+	fileToIndex map[string]int
```

Comment:
Why do you need to store this on struct? Especially unexported?

Reply:
Convenience. Otherwise you need to pass this structure through function args and return values.